integrationtests/test_all_examples.py

```diff
@@ -307,6 +308,43 @@ def load_demo(file_path, variable_name="demo"):
     return globals_[variable_name]
 
 
+# =============================================================================
+# load_tests protocol for unittest discover
+# =============================================================================
+
+
+def load_tests(loader, tests, pattern):
+    """ Implement load_tests protocol so that when unittest discover is run
+    with this test module, the tests in the demo folder (not a package) are
+    also loaded.
+
+    See unittest documentation on load_tests
+    """
+    # Keep all the other loaded tests.
+    suite = unittest.TestSuite()
+    suite.addTests(tests)
+
+    # Expand the test suite with tests from the examples, assuming
+    # the test for ``group/script.py`` is placed in ``group/tests/`` directory.
+    accepted_files, _ = SEARCHER.get_python_files()
+    test_dirs = set(
+        os.path.join(os.path.dirname(path), "tests") for path in accepted_files
+    )
+    test_dirs = set(path for path in test_dirs if os.path.exists(path))
+    for dirpath in sorted(test_dirs):
+
+        # Test files are scripts too and they demonstrate running the
+        # tests. Mock the run side-effect when we load the test cases.
+        with mock.patch.object(unittest.TextTestRunner, "run"):
```

Comment:
This is mocking the `run` invocation here:
https://github.com/enthought/traitsui/blob/7b0a8c7122fa6995afdaea56349731d1ea77b56d/traitsui/examples/demo/Standard_Editors/tests/test_TextEditor_demo.py#L59-L62

This is not wrapped under `if __name__ == "__main__"` so that it can be run from the etsdemo.

In order to we load the test case, we need to load the file, so this mock prevents the test from being run when we are just loading the test case.